--- a/drivers/md/Makefile
+++ b/drivers/md/Makefile
@@ -27,7 +27,11 @@
 dm-verity-y	+= dm-verity-target.o
 dm-zoned-y	+= dm-zoned-target.o dm-zoned-metadata.o dm-zoned-reclaim.o
 
-md-mod-y	+= md.o md-bitmap.o
+ifeq ($(CONFIG_MD_UNRAID),y)
+md-mod-y        += md_unraid.o unraid.o
+else
+md-mod-y        += md.o md-bitmap.o
+endif
 raid456-y	+= raid5.o raid5-cache.o raid5-ppl.o
 linear-y	+= md-linear.o
 multipath-y	+= md-multipath.o
@@ -48,9 +52,11 @@
 obj-$(CONFIG_MD_CLUSTER)	+= md-cluster.o
 obj-$(CONFIG_BCACHE)		+= bcache/
 obj-$(CONFIG_BLK_DEV_MD)	+= md-mod.o
+ifneq ($(CONFIG_MD_UNRAID),y)
 ifeq ($(CONFIG_BLK_DEV_MD),y)
 obj-y				+= md-autodetect.o
 endif
+endif
 obj-$(CONFIG_BLK_DEV_DM)	+= dm-mod.o
 obj-$(CONFIG_BLK_DEV_DM_BUILTIN) += dm-builtin.o
 obj-$(CONFIG_DM_UNSTRIPED)	+= dm-unstripe.o
