--- a/drivers/nvme/host/pci.c-	2022-09-05 13:48:16.267794964 -0700
+++ b/drivers/nvme/host/pci.c	2022-09-05 13:49:12.754243403 -0700
@@ -3480,6 +3480,9 @@
 	{ PCI_DEVICE(0x10ec, 0x5762),   /* ADATA SX6000LNP */
 		.driver_data = NVME_QUIRK_IGNORE_DEV_SUBNQN |
 				NVME_QUIRK_BOGUS_NID, },
+	{ PCI_DEVICE(0x10ec, 0x5763),   /* limetech: Team Group MP33 M.2 2280 1TB NVMe */
+		.driver_data = NVME_QUIRK_IGNORE_DEV_SUBNQN |
+				NVME_QUIRK_BOGUS_NID, },
 	{ PCI_DEVICE(0x1cc1, 0x8201),   /* ADATA SX8200PNP 512GB */
 		.driver_data = NVME_QUIRK_NO_DEEPEST_PS |
 				NVME_QUIRK_IGNORE_DEV_SUBNQN, },
