From 45cf2e2ed1d500e8dcf868f3e3d43c668a660730 Mon Sep 17 00:00:00 2001
From: pieceking <1@1.com>
Date: Thu, 29 Sep 2022 22:51:06 +0800
Subject: [PATCH 1/6] rename application package name

---
 app/build.gradle | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/app/build.gradle b/app/build.gradle
index d756828..b0dc746 100644
--- a/app/build.gradle
+++ b/app/build.gradle
@@ -4,7 +4,7 @@ android {
     compileSdkVersion 30
 
     defaultConfig {
-        applicationId 'com.github.tvbox.osc'
+        applicationId 'com.github.tvbox.osc.muzilingql'
         minSdkVersion 16
         targetSdkVersion 26
         versionCode 1
-- 
2.37.3.windows.1

