From 1a17c0b3f962e51bd6bfa62ee118487b69241d37 Mon Sep 17 00:00:00 2001
From: pieceking <1@1.com>
Date: Mon, 19 Sep 2022 23:40:17 +0800
Subject: [PATCH 1/6] rename application package name

---
 app/build.gradle | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/app/build.gradle b/app/build.gradle
index f9e31ce..e092307 100644
--- a/app/build.gradle
+++ b/app/build.gradle
@@ -4,7 +4,7 @@ android {
     compileSdkVersion 30
 
     defaultConfig {
-        applicationId 'com.github.tvbox.osc.tk'
+        applicationId 'com.github.tvbox.osc.muziling'
         minSdkVersion 16
         targetSdkVersion 26
         versionCode 1
-- 
2.37.1.windows.1

